src/imports/torusVerifier.js

```diff
@@ -0,0 +1,90 @@
+import FetchNodeDetails from '@toruslabs/fetch-node-details/dist/fetchNodeDetails-node.js'
+import TorusUtils from '@toruslabs/torus.js/dist/torusUtils-node.js'
+import Config from '../server/server.config'
+import { recoverPublickey } from '../server/utils/eth'
+class TorusVerifier {
+  constructor(proxyContract, network) {
+    this.fetchNodeDetails = new FetchNodeDetails({ network, proxyAddress: proxyContract })
+    this.torus = new TorusUtils()
+  }
+
+  async isIdentifierOwner(publicAddress, verifier, identifier) {
+    const { torusNodeEndpoints, torusNodePub } = await this.fetchNodeDetails.getNodeDetails()
+    const response = await this.torus.getPublicAddress(
+      torusNodeEndpoints,
+      torusNodePub,
+      { verifier, verifierId: identifier },
+      false
+    )
+    return publicAddress.toLowerCase() === response.toLowerCase()
+  }
+
+  getVerifierAndIdentifier(torusType, userRecord) {
+    switch (torusType) {
+      case 'google-old':
+        return {
+          verifier: 'google-gooddollar',
+          identifier: userRecord.email,
+          emailVerified: true,
+          mobileVerified: false
+        }
+      case 'google':
+        return {
+          verifier: 'google-auth0-gooddollar',
+          identifier: userRecord.email,
+          emailVerified: true,
+          mobileVerified: false
+        }
+      case 'facebook':
+        return {
+          verifier: 'facebook-gooddollar',
+          identifier: userRecord.email,
+          emailVerified: true,
+          mobileVerified: false
+        }
+      case 'auth0-pwdless-email':
+        return {
+          verifier: 'google-auth0-gooddollar',
+          identifier: userRecord.email,
+          emailVerified: true,
+          mobileVerified: false
+        }
+      case 'auth0-pwdless-sms':
+        return {
+          verifier: 'google-auth0-gooddollar',
+          identifier: userRecord.mobile,
+          emailVerified: false,
+          mobileVerified: true
+        }
+      default:
+        throw new Error('unknown torus login type: ' + torusType)
+    }
+  }
+  async isFacebookEmail(token, email) {
+    return true
+  }
+  async verifyProof(signature, torusType, userRecord, nonce) {
+    if (Date.now() - nonce > 60000) {
```

Comment:
we're using moment so we could write this in more readable way:
```
if (moment().diff(moment(nonce), 'minutes') >= 1) {
```

Reply:
done